### PR TITLE
Feature etb#190 uart seds

### PR DESCRIPTION
**Pull request by jrachucki-n7s**

### CODE REVIEW

File: Drivers/seds/UartHwas/example/test_binaries.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	pushd samv71Partition/work/binaries > /dev/null || exit
+
+runSamV71Binary.py partition_1 -u /tmp/serialTest -v vConsole> /dev/null & pid1=$!
+
+while [ ! -L "/tmp/serialTest" ]
+do
+    sleep 1
+done
+
+echo "PTYs created, let's rock"
```
Comment (Lurkerpas) [inline Drivers/seds/UartHwas/example/test_binaries.sh]: Please remove "let's rock" :)

File: Drivers/seds/UartHwas/UartHwas.xml [hunk 2]
```diff
@@ -0,0 +1,1330 @@
...
+          <Command name="SendByteAsyncCmd" mode="async">
+            <LongDescription>Sends one byte asynchroniously</LongDescription>
+            <Argument name="uart" type="UartHwas" mode="in" />
+            <Argument name="byteToSend" type="com/n7space/Hwas/Word" mode="in" />
+            <Argument name="byteSentFlag" type="com/n7space/Hwas/Word" mode="notify" />
+          </Command>
+        </CommandSet>
+      </Interface>
+    </DeclaredInterfaceSet>
+    <ComponentSet>
```
Comment (Lurkerpas) [inline Drivers/seds/UartHwas/UartHwas.xml:80]: Will this get split in next MR?

File: Drivers/seds/UartHwas/UartHwas.xml [hunk 3]
```diff
@@ -0,0 +1,1330 @@
...
+              </Body>
+            </Activity>
+            <Activity name="SendByteAsync">
+              <Argument name="uart" type="UartHwas"></Argument>
+              <Argument name="byte" type="com/n7space/Hwas/Word"></Argument>
+              <Body>
+                <Assignment outputVariableRef="irqNumberVar">
+                  <VariableRef variableRef="uartVar.mIrqNumber" />
+                </Assignment>
+                <SendCommandPrimitive shortDescription="Subscribe to Uart interrupt" interface="InterruptSubscriptionManagement" command="SubscribeToInterrupt">
```
Comment (Lurkerpas) [inline Drivers/seds/UartHwas/UartHwas.xml]: Does this call needs repetition?

File: Drivers/seds/UartHwas/UartHwas.xml [hunk 4]
```diff
@@ -0,0 +1,1330 @@
...
+                            </Conditional>
+                          </OnConditionFalse>
+                        </Conditional>
+                      </OnConditionFalse>
+                    </Conditional>
+                  </OnConditionFalse>
+                </Conditional>
+              </Body>
+            </Activity>
+            <Activity name="GetLocalUart">
```
Comment (Lurkerpas) [inline Drivers/seds/UartHwas/UartHwas.xml:201]: Let's discuss how this works

File: Drivers/seds/UartHwas/UartHwas.xml (hunk 2 above)
Comment (jrachucki-n7s) [reply Drivers/seds/UartHwas/UartHwas.xml:80]: Yes, it will.

File: Drivers/seds/UartHwas/UartHwas.xml (hunk 4 above)
Comment (jrachucki-n7s) [reply Drivers/seds/UartHwas/UartHwas.xml:201]: It works in the same way as https://github.com/n7space/TASTE-SAMV71-Runtime/blob/main/src/Hal/Hal.c.
The interrupt handler needs the reference to the UART that triggered that interrupt. It needs it as it must read the values from a couple of registers. I have added the UART variables, an equivalent of the static uartXHandles from the example above, to pass correct register addresses to the interrupt handler.

File: Drivers/seds/UartHwas/example/test_binaries.sh (hunk 1 above)
Comment (jrachucki-n7s) [reply Drivers/seds/UartHwas/example/test_binaries.sh]: Done

File: Drivers/seds/UartHwas/UartHwas.xml (hunk 3 above)
Comment (jrachucki-n7s) [reply Drivers/seds/UartHwas/UartHwas.xml]: It is not needed. I have deleted it.

**Review by Lurkerpas — APPROVED** [review]

**Pull request merged**